finality-aleph/src/party/manager/task.rs

```diff
@@ -12,26 +12,43 @@ pub type Handle = Pin<Box<(dyn Future<Output = sc_service::Result<(), ()>> + Sen
 pub struct Task {
     handle: Handle,
     exit: oneshot::Sender<()>,
+    error_on_exit: Option<bool>,
```

Comment:
Suggesion: I'm confused by this three-state bool here, so I would propose to change it to some enum with informational variant names. Maybe `status: Running/ExitedWithError/ExitedCleanly`? As I said I'm confused, so not sure what exact information this is supposed to convey, so my proposed names may be way off.

Reply:
changed its type to `Option<Result<(), ()>>` and renamed to `cached_result`